caddyhttp/proxy/proxy.go

```diff
@@ -255,6 +255,11 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 			return http.StatusRequestEntityTooLarge, backendErr
 		}
 
+		// TODO: write an 499 HTTP status as constant in someplace (and also change in proxy_test.go on line: expectedStatus, expectErr := 499, context.Canceled)
```

Comment:
I guess we can start by putting the `const StatusContextCancelled = 499` at the bottom of this file

Reply:
Ok. But I suggest a `CustomStatusContextCancelled` because this is not a standard HTTP status. Are you agreed?

Reply:
Done.

Reply:
@mholt  Are you happy with this change.  Can you approve and I will merge.  @francislavoie would you like to approve changes also.